src/plugins/pcr/src/InSilicoPcrTask.cpp

```diff
@@ -107,10 +107,24 @@ FindAlgorithmTaskSettings InSilicoPcrTask::getFindPatternSettings(U2Strand::Dire
 }
 
 void InSilicoPcrTask::prepare() {
+    if (!PrimerStatistics::validatePrimerMinLength(settings.forwardPrimer) || !PrimerStatistics::validatePrimerMinLength(settings.reversePrimer)) {
+        algoLog.details(tr("One of the given primers is too short. Task cancelled."));
+        stateInfo.setCanceled(true);
```

Comment:
This should be failed, not canceled

+ Why don't we check these values in the dialog, before running the task?

Reply:
What dialog do you mean?
Entered settings already checked in OP tab.

Reply:
> Entered settings already checked in OP tab.

OP tab has no check for the length of the primers. It means that I can run primers search task even with primers like 'A'  and 'A' which is wrong (and will make UGENE hang).

Safety checks in tasks are good, but we should not even allow running primer search algorithm for invalid input.

Reply:
Any updates for this part?